### PR TITLE
Add EditMeeting Command

### LINKED ISSUE #76: Add EditMeetingCommand

### PR DESCRIPTION
**Pull request by psh12320**
fixes #76

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 1]
```diff
@@ -0,0 +1,260 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
```
Comment (marraknas) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:3]: Once again same as DeleteMeeting, need test cases for codecov!

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        // Is a similar update meeting list method required? Ask peter TODO
+
+        return new CommandResult(String.format(MESSAGE_EDIT_MEETING_SUCCESS, editedMeeting));
+    }
+
+    /**
+     * Creates and returns a {@code Meeting} with the details of {@code meetingToEdit}
+     * edited with {@code editMeetingDescriptor}.
+     */
+    private static Meeting createEditedMeeting(Meeting meetingToEdit, EditMeetingDescriptor editMeetingDescriptor)
```
Comment (marraknas) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:130]: good method!

File: src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
```
Comment (marraknas) [inline src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java:3]: Test cases here also!

**Review by marraknas — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/address/logic/commands/DeleteMeetingCommand.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
+package seedu.address.logic.commands;
```
Comment (PeterHW963) [inline src/main/java/seedu/address/logic/commands/DeleteMeetingCommand.java:1]: thinking....

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 5]
```diff
@@ -79,6 +79,7 @@ public CommandResult execute(Model model) throws CommandException {
         List<Person> lastShownList = model.getFilteredPersonList();
```
Comment (PeterHW963) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:79]: thinking....

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 6]
```diff
@@ -0,0 +1,260 @@
+package seedu.address.logic.commands;
```
Comment (PeterHW963) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:1]: thinking....

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java (hunk 2 above)
Comment (PeterHW963) [reply src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:130]: thinking....

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 7]
```diff
@@ -12,7 +12,9 @@
 import seedu.address.logic.commands.ClearCommand;
 import seedu.address.logic.commands.Command;
```
Comment (PeterHW963) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:13]: thinking....

File: src/main/java/seedu/address/logic/parser/DeleteMeetingCommandParser.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package seedu.address.logic.parser;
```
Comment (PeterHW963) [inline src/main/java/seedu/address/logic/parser/DeleteMeetingCommandParser.java:1]: thinking....

File: src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_END_TIME;
```
Comment (PeterHW963) [inline src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java:5]: thinking....

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 10]
```diff
@@ -2,6 +2,9 @@
 
 import static java.util.Objects.requireNonNull;
 
+import java.time.LocalDateTime;
```
Comment (PeterHW963) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:5]: thinking....

File: src/main/java/seedu/address/model/AddressBook.java [hunk 11]
```diff
@@ -125,10 +125,40 @@ public void addMeeting(Meeting m) {
         meetings.addMeeting(m);
     }
 
+    public void deleteMeeting(Meeting m) {
```
Comment (PeterHW963) [inline src/main/java/seedu/address/model/AddressBook.java:128]: thinking....

File: src/main/java/seedu/address/model/Model.java [hunk 12]
```diff
@@ -92,5 +92,31 @@ public interface Model {
      */
     void addMeeting(Person target, Meeting meeting) throws CommandException;
 
+    /**
+     * Deletes a given meeting with a person.
+     */
+    void deleteMeeting(Person target, Meeting meeting);
```
Comment (PeterHW963) [inline src/main/java/seedu/address/model/Model.java:98]: thinking....

File: src/main/java/seedu/address/model/ModelManager.java [hunk 13]
```diff
@@ -115,14 +115,48 @@ public void setPerson(Person target, Person editedPerson) {
 
     @Override
     public void addMeeting(Person target, Meeting meeting) {
-        target.getMeetings().addMeeting(meeting);
+        // Change in main meeting list first to check for any time clash exception in all meetings
```
Comment (PeterHW963) [inline src/main/java/seedu/address/model/ModelManager.java:118]: thinking....

File: src/main/java/seedu/address/model/person/Meetings.java [hunk 14]
```diff
@@ -1,13 +1,16 @@
 package seedu.address.model.person;
```
Comment (PeterHW963) [inline src/main/java/seedu/address/model/person/Meetings.java:1]: thinking....

File: src/main/java/seedu/address/model/person/exceptions/DuplicateMeetingException.java [hunk 15]
```diff
@@ -0,0 +1,11 @@
+package seedu.address.model.person.exceptions;
```
Comment (PeterHW963) [inline src/main/java/seedu/address/model/person/exceptions/DuplicateMeetingException.java:1]: I'll stop here. My brain doesn't want to think anymore

**Review by PeterHW963 — COMMENTED** [review]
LMTM (Let me think more)!

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 16]
```diff
@@ -0,0 +1,260 @@
...
+
+        if (index.getZeroBased() >= model.getMeetingSize()) {
+            System.out.println("INDEX VALUE: " + index.getZeroBased() + ", MEETING SIZE: " + model.getMeetingSize());
+            throw new CommandException(Messages.MESSAGE_INVALID_MEETING_INDEX);
+        }
+
+        Meeting meetingToEdit = model.getMeeting(index.getZeroBased());
+        Meeting editedMeeting = createEditedMeeting(meetingToEdit, editMeetingDescriptor);
+
+        // Check if the new name exists or not
```
Comment (XinghuaJulia) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:78]: Are you sure you want to edit the name? KIV (maybe?)

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 17]
```diff
@@ -0,0 +1,260 @@
...
+                // Different person; delete for old, add for new
+                model.deleteMeeting(personBeingEdited, meetingToEdit);
+                model.addMeeting(personToEditMeeting, editedMeeting);
+            }
+        } catch (MeetingNotFoundException | TimeClashException e) {
+            throw new CommandException(e.getMessage());
+        }
+
+        // model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        // Is a similar update meeting list method required? Ask peter TODO
```
Comment (XinghuaJulia) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:121]: Urm what??? Ask Peter TODO wha??

**Review by XinghuaJulia — COMMENTED** [review]
LGTM, minor fixes you may want to look at in terms of task delegation...

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 18]
```diff
@@ -0,0 +1,257 @@
...
+                personBeingEdited = person;
+            }
+        }
+
+        Person personToEditMeeting = null;
+        boolean personExists = false;
+        for (Person person : model.getFilteredPersonList()) {
+            if (person.getName().equals(newName)) {
+                personExists = true;
+                personToEditMeeting = person;
```
Comment (ulricolo7) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:92]: It's great that you are considering the potential bugs of editing the names in the meetings!

**Review by ulricolo7 — COMMENTED** [review]
LGTM! Thanks Shri!

**Pull request merged**